cpp/src/barretenberg/dsl/acir_format/keccak_constraint.hpp

```diff
@@ -0,0 +1,52 @@
+#pragma once
+#include <cstdint>
+#include <vector>
+#include "barretenberg/dsl/types.hpp"
+
+namespace acir_format {
+
+struct HashInput {
```

Comment:
As we are going to be extending this to be used by other hash types it could be useful to have it in a separate file. A new `hash_input.hpp` file feels unnecessary though and it can be a part of `acir_format.hpp`. However, this is not blocking imo as it is only used by Keccak at the moment, just leaving the note for visibility.

Reply:
Good point, I will do this when it is used by other hashes.